### PR TITLE
Add ci for nestest

### PR DESCRIPTION
**Pull request by johnor**
Had planned to run with sanitizers enabled at first but haven't been able to get to symbols to show correctly in the stacktrace on github actions yet. We can enable that later.

### CODE REVIEW

File: .github/workflows/ci-nestest.yaml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+      - name: Build
+        run: |
+          cmake --build build --config ${{ matrix.configuration }} --target nestest
+
+      - name: Run Nestest
+        run: |
+          wget http://nickmass.com/images/nestest.nes
+          wget http://www.qmtpro.com/~nes/misc/nestest.log
+          ./build/nestest/nestest nestest.nes | tee nestest-output.log
```
Comment (robinlinden) [inline .github/workflows/ci-nestest.yaml:54]: Maybe run diff on the 2 logs too? That way we don't have to manually download and diff them like peasants. 😁

Comment (johnor) [reply .github/workflows/ci-nestest.yaml:54]: That is what the python script it for :) Was planning on getting this merged first and then update once the script is merged. Unfortunately we can't just diff as our ppu counter is a bit different.

Comment (johnor) [reply .github/workflows/ci-nestest.yaml:54]: Actually, will fix now since it was approved already :)

**Comment by johnor** [comment]
The step "Diff output" will show the first incorrect instruction now. The return code is ignored though as we have not implemented all instructions yet.

**Pull request merged**